core/src/main/java/org/hisp/dhis/android/core/period/clock/internal/ClockProviderFactory.kt

```diff
@@ -25,31 +25,26 @@
  *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+package org.hisp.dhis.android.core.period.clock.internal
 
-package org.hisp.dhis.android.core.period.internal;
+import kotlinx.datetime.LocalDateTime
 
-import org.hisp.dhis.android.core.period.PeriodType;
+internal object ClockProviderFactory {
 
-import java.util.Calendar;
+    var clockProvider: ClockProvider = RegularClockProvider()
+        private set
 
-class DailyPeriodGenerator extends AbstractPeriodGenerator {
-
-    DailyPeriodGenerator(Calendar calendar) {
-        super(calendar, "yyyyMMdd", PeriodType.Daily);
-    }
-
-    @Override
-    protected void moveToStartOfCurrentPeriod() {
-        // do nothing
+    fun setRegular() {
+        clockProvider = RegularClockProvider()
     }
 
-    @Override
-    protected void moveToStartOfCurrentYear() {
-        calendar.set(Calendar.DAY_OF_YEAR, 1);
+    fun setFixed() {
```

Comment:
This function and the one below seem to be devoted for testing only. Couldn't they be moved to a test helper? The fixed date below seems a bit arbitrary if left here. Also, the original one sets 11 for the month instead of 12, but it probably doesn't matter.

Reply:
The original month set to 11 actually was December (zero-based). But the day was 1-based. It is one of the tricky things about Calendar.

Makes sense to move it to a test helper, I will do it